tests/test_core.py

```diff
@@ -23,15 +24,112 @@ class test_PairMatrix(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different list types."""
-        raise NotImplementedError()
+
+        pair_types = ('A','B')
+        pair_dict = {('A','B'):{}, ('B','B'):{}, ('A','A'):{}}
+
+        #test construction with tuple input
+        m = relentless.core.PairMatrix(types=('A','B'))
+        self.assertEqual(m.types, pair_types)
+        self.assertEqual(m._data, pair_dict)
```

Comment:
I wouldn't test `m._data` directly in case the internal storage type changes later. Just check `m.pairs` for now, and then the `test_accessors` will cover the internals.